pkg/config/polling_manager.go

```diff
@@ -157,21 +152,9 @@ func (cm *PollingProjectConfigManager) SyncConfig(datafile []byte) {
 		return
 	}
 	cmLogger.Debug(fmt.Sprintf("New datafile set with revision: %s. Old revision: %s", projectConfig.GetRevision(), previousRevision))
-	cm.projectConfig = projectConfig
-	if cm.optimizelyConfig != nil {
-		cm.optimizelyConfig = NewOptimizelyConfig(projectConfig)
-	}
+	cm.setConfig(projectConfig)
 	closeMutex(nil)
-
-	if cm.notificationCenter != nil {
-		projectConfigUpdateNotification := notification.ProjectConfigUpdateNotification{
-			Type:     notification.ProjectConfigUpdate,
-			Revision: cm.projectConfig.GetRevision(),
-		}
-		if err = cm.notificationCenter.Send(notification.ProjectConfigUpdate, projectConfigUpdateNotification); err != nil {
-			cmLogger.Warning("Problem with sending notification")
-		}
-	}
+	cm.sendConfigUpdateNotification()
```

Comment:
I think cm.setConfig() should still return an error, and we need to sendConfigUpdateNotification only when error is not nil